src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,192 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import exception.DukeException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+
+
+/**
+ * Class that handles userInput and return appropriate action.
+ *
+ * @author syamfarh
+ */
+public class Parser {
+
+    /**
+     * parse UserInput and passed the result to ui to output the result.
+     *
+     * @param input User input from main.
+     * @param tasks current Tasklist from chatBot.
+     * @param ui ui that control output result to user.
+     */
+    public static void replyUser(String input, TaskList tasks, Ui ui) throws DukeException {
+
+        switch (input) {
+        case "bye":
+            ui.exitGreeting();
+            break;
+        case "barbie":
+            ui.customReply();
+            break;
+        case "list":
+            ui.outputList(tasks);
+            break;
+        default:
```

Comment:
maybe the contents of this could make use of some abstraction and separate if statements to avoid deep nesting?